File: scribbleshare-frontend/src/scripts/protocol/server/messages/ServerMessageMouseMove.js
import ServerMessage from "../ServerMessage.js";
import ServerMessageType from "../ServerMessageType.js";
import MouseMove from "../../../MouseMove.js";

export default class ServerMessageMouseMove extends ServerMessage {
    constructor(reader) {
        super(ServerMessageType.MOUSE_MOVE);
        this.client = reader.readInt16();
        this.mouseMoves = [];
        let length = reader.readUint8();
        for (let i = 0; i < length; i++) {
            this.mouseMoves[i] = new MouseMove(reader);
        }
    }

}

File: scribbleshare-room/src/main/java/net/stzups/scribbleshare/room/server/websocket/protocol/server/messages/ServerMessageMouseMove.java
package net.stzups.scribbleshare.room.server.websocket.protocol.server.messages;

import io.netty.buffer.ByteBuf;
import net.stzups.scribbleshare.room.server.websocket.Client;
import net.stzups.scribbleshare.room.server.websocket.MouseMove;
import net.stzups.scribbleshare.room.server.websocket.protocol.server.ServerMessage;
import net.stzups.scribbleshare.room.server.websocket.protocol.server.ServerMessageType;

public class ServerMessageMouseMove extends ServerMessage {
    private final Client client;
    private final MouseMove[] mouseMoves;

    public ServerMessageMouseMove(Client client, MouseMove[] mouseMoves) {
        super(ServerMessageType.MOUSE_MOVE);
        this.client = client;
        this.mouseMoves = mouseMoves;
    }

    @Override
    public void serialize(ByteBuf byteBuf) {
        super.serialize(byteBuf);
        client.serialize(byteBuf);
        byteBuf.writeByte((byte) mouseMoves.length);
        for (MouseMove mouseMove : mouseMoves) {
            mouseMove.serialize(byteBuf);
        }
    }
}
